# CSE-221710313059-T.Yasasvi

In [1]:
import pandas as pd
import numpy as np

In [3]:
data=pd.read_csv("https://raw.githubusercontent.com/Mounika-Kajjam/Datasets/master/cities.csv")
data.head()

name_of_city  state_code      state_name  dist_code  population_total  \
0      Abohar            3          PUNJAB          9            145238   
1    Achalpur           27     MAHARASHTRA          7            112293   
2    Adilabad           28  ANDHRA PRADESH          1            117388   
3   Adityapur           20       JHARKHAND         24            173988   
4       Adoni           28  ANDHRA PRADESH         21            166537   

   population_male  population_female  0-6_population_total  \
0            76840              68398                 15870   
1            58256              54037                 11810   
2            59232              58156                 13103   
3            91495              82493                 23042   
4            82743              83794                 18406   

   0-6_population_male  0-6_population_female  ...  literates_female  \
0                 8587                   7283  ...             44972   
1                 6186                   5624  ...             43086   
2                 6731                   6372  ...             37660   
3                12063                  10979  ...             54515   
4                 9355                   9051  ...             45089   

   sex_ratio  child_sex_ratio  effective_literacy_rate_total  \
0        890              848                          79.86   
1        928              909                          91.99   
2        982              947                          80.51   
3        902              910                          83.46   
4       1013              968                          68.38   

   effective_literacy_rate_male  effective_literacy_rate_female  \
0                         85.49                           73.59   
1                         94.77                           89.00   
2                         88.18                           72.73   
3                         89.98                           76.23   
4                         76.58                           60.33   

                location  total_graduates male_graduates  female_graduates  
0  30.1452928,74.1993043            16287           8612              7675  
1   21.257584,77.5086754             8863           5269              3594  
2   19.0809075,79.560344            10565           6797              3768  
3  22.7834741,86.1576889            19225          12189              7036  
4  15.6322227,77.2728368            11902           7871              4031  

[5 rows x 22 columns]

In [4]:
#a.Analyze the  top 5 cities with highest number of female graduates
df1=data.sort_values(by='female_graduates',ascending=False).head(5)
df1['name_of_city']

141                Delhi 
185       Greater Mumbai 
72              Bengaluru
184    Greater Hyderabad 
119              Chennai 
Name: name_of_city, dtype: object

In [5]:
#b.Find the 5 cities with lowest number of female graduates
#and concatenate the two dataframes of top 5 cities with  bottom 5 cities
df2=data.sort_values(by='female_graduates',ascending=True).head(5)
df2['name_of_city']

131       Dabgram 
39         Bagaha 
234       Jamuria 
374     Pithampur 
323    Mustafabad 
Name: name_of_city, dtype: object

In [6]:
frames=[df1,df2]
pd.concat(frames)

name_of_city  state_code      state_name  dist_code  \
141              Delhi            7    NCT OF DELHI         99   
185     Greater Mumbai           27     MAHARASHTRA         99   
72            Bengaluru          29       KARNATAKA         18   
184  Greater Hyderabad           28  ANDHRA PRADESH         99   
119            Chennai           33      TAMIL NADU          2   
131            Dabgram           19     WEST BENGAL          2   
39              Bagaha           10           BIHAR          1   
234            Jamuria           19     WEST BENGAL          9   
374          Pithampur           23  MADHYA PRADESH         21   
323         Mustafabad            7    NCT OF DELHI          3   

     population_total  population_male  population_female  \
141          11007835          5871362            5136473   
185          12478447          6736815            5741632   
72            8425970          4401299            4024671   
184           6809970          3500802            3309168   
119           4681087          2357633            2323454   
131            118464            60708              57756   
39             113012            59946              53066   
234            144971            75574              69397   
374            126099            70577              55522   
323            127012            66754              60258   

     0-6_population_total  0-6_population_male  0-6_population_female  ...  \
141               1209275               647938                 561337  ...   
185               1139146               599007                 540139  ...   
72                 862493               444639                 417854  ...   
184                725816               373794                 352022  ...   
119                418541               213084                 205457  ...   
131                 14472                 7427                   7045  ...   
39                  19070                 9802                   9268  ...   
234                 17556                 9017                   8539  ...   
374                 20438                10872                   9566  ...   
323                 19370                10176                   9194  ...   

     literates_female  sex_ratio  child_sex_ratio  \
141           3806615        875              866   
185           4509812        852              902   
72            3110983        914              940   
184           2359594        945              942   
119           1845974        986              964   
131             34149        951              949   
39              22247        885              946   
234             38753        918              947   
374             33387        787              880   
323             34201        903              903   

     effective_literacy_rate_total  effective_literacy_rate_male  \
141                          87.60                         91.44   
185                          90.28                         93.32   
72                           89.59                         92.63   
184                          82.96                         85.96   
119                          90.33                         93.47   
131                          73.98                         80.29   
39                           60.67                         69.30   
234                          73.11                         81.73   
374                          82.73                         90.49   
323                          76.10                         84.33   

     effective_literacy_rate_female               location  total_graduates  \
141                           83.20        28.7041,77.1025          2221137   
185                           86.70        19.0760,72.8777          1802371   
72                            86.25        12.9716,77.5946          1591163   
184                           79.79        17.3850,78.4867          1164149   
119                      

In [7]:
#d.Find the  top 10 cities with highest total graduation rate
df3=data.sort_values(by='total_graduates',ascending=False).head(10)
df3['name_of_city']

141                Delhi 
185       Greater Mumbai 
72              Bengaluru
184    Greater Hyderabad 
119              Chennai 
274              Kolkata 
7              Ahmadabad 
380                 Pune 
288              Lucknow 
225               Jaipur 
Name: name_of_city, dtype: object

In [10]:
#e.Calculate the average child sex ratio of the Kerala State.
df4=data.groupby('state_name')['sex_ratio'].mean()
df4['KERALA']

1066.7142857142858

2.Q.Mean = 70 and Standard Deviation = 10. Identify how much area can be found between 70 and 90 with help of Empirical Rule.<br>
Sol: 60-80 covers 68% of the data<br>
50-90 covers 95% of the data, since a normal distribution curve is symmetric 70-90 contains half of 50-90.
Therefore, 70-70 covers 47.5% data